### DISCUSSION TITLE
Option to apply individual settings globally or on a per-site basis

### DISCUSSION THREAD

**Discussion by cmurtheepic** [dhowe/AdNauseam#2558] [Ideas]
We really need a way to tell Adnauseam that we want a setting, or settings, to apply and not apply to every website.

Such as:
- The image size limit that breaks a large number of websites.
- Click Ads
- And other settings.

We really just need a way to set certain, or all settings to apply globally (meaning every website) or on a per-site basis.

**Comment by dhowe** [comment]
Pls create an issue for this: "The image size limit that breaks a large number of websites", including one or more example sites...
thanks

**Reply by cmurtheepic** [reply]
Sure. But it literally effects ANY website with images on it... it will atleast break one image, if not every image on the website. 

I'm sure you're aware of this as a maintainer of the project.

**Reply by dhowe** [reply]
> I'm sure you're aware of this as a maintainer of the project.

We are not. What is the image size you are using? And have you tried toggling off this function ?

**Reply by cmurtheepic** [reply]
> > I'm sure you're aware of this as a maintainer of the project.
>
>We are not. What is the image size you are using? And have you tried toggling off this function?
>
@dhowe
Sorry, I'm not trying to be rude, but I am genuinely flabbergasted as to how that could be the case. I am a Cisco-certified IT technician. I had my coding days, and I still keep up with it all as that is part of my job. And as such I still have a good grasp of how any form of coding works, how things are implemented, and I can still read code for the most part. So, I'm genuinely curious as to how a setting that has the obvious potential to break most websites with the current web development techniques such as copious amounts of background images, product images, parallax effects, and other forms of imagery that a majority of, are above the 50 KB block default; which breaks or disrupts nearly every site I go to outside of the top websites. Maybe it's because it is not recognized as a problem, and users think it's a feature for websites to look like that. Maybe it's down to how few people there are using Adnauseam, and perhaps their various use cases don't end up with them running into those situations. Or just plain underreported.


"What is the image size you are using?"
I'm just using the Adnauseam default which is "50 KB" on every site, unless you turn it off. (Circled in Red)
"And have you tried toggling off this function?"
Yes, and you can only turn it off on a per-site basis. (Circled in Blue)

Meaning, I first have to visit the website.

Notice that images aren't loading, or the layout is messed up by the images not loading.

Then go into Adnauseam settings and have to disable this every time. (Circled in Red) If not disable Adnauseam for that website entirely.
<img width="390" alt="Block Media Elements greater than a certain size Adnauseam bug" src="https://github.com/user-attachments/assets/ea5145c2-adf1-490a-83a8-6b5fae6e19ba">

**Reply by mneunomne** [reply]
@cmurtheepic can you please share the content inside your "My Rules" tab in settings?

**Reply by cmurtheepic** [reply]
> @cmurtheepic can you please share the content inside your "My Rules" tab in settings?

@mneunomne 
Sure, but I had to reinstall it to upgrade Adnauseam to the newest stable build. Which reset all of my settings and configuration, so there's not much to see at the moment.
<img width="1919" alt="Adnauseam My Rules Screenshot" src="https://github.com/user-attachments/assets/e5ca1ecf-d3b6-4bea-bd38-b420d890035f">

**Reply by cmurtheepic** [reply]
@dhowe @mneunomne And as far as I can tell the app keeps deleting my trusted sites everyday. And my educated guess says it's probably resetting itself; either every time I re-open chrome, or on computer power off, and I haven't had time to test my theory. But it is definitely resetting my trusted sites page atleast once a day. I think I have seen other settings get turned back on but I haven't played enough attention to notice since my trusted sites being broken Is much more readily apparent.

**Comment by dhowe** [comment]
can you try the same test with a fresh browser profile ?

**Reply by cmurtheepic** [reply]
@dhowe I will update this when I get around to reinstalling windows 10 on my pc, which badly needs to be done first as well.
